### ISSUE TITLE
[CI] RecoveryRequestTrackerTests.testIdempotencyIsEnforced failed on 7.x

### ISSUE THREAD

**Issue by nknize** [elastic/elasticsearch#58627]
Build: https://gradle-enterprise.elastic.co/s/st676wtk24nle

Reproduces with:

```
./gradlew ':server:test' --tests "org.elasticsearch.indices.recovery.RecoveryRequestTrackerTests.testIdempotencyIsEnforced" 
-Dtests.seed=5BBD0021C5179A42 
-Dtests.security.manager=true 
-Dtests.locale=es-CR 
-Dtests.timezone=Africa/Banjul 
-Druntime.java=8 
-Dtests.fips.enabled=true
```

Error:
```
14:47:50 org.elasticsearch.indices.recovery.RecoveryRequestTrackerTests > testIdempotencyIsEnforced FAILED
14:47:50     java.lang.AssertionError
14:47:50         at __randomizedtesting.SeedInfo.seed([5BBD0021C5179A42:99253880B16D0044]:0)
14:47:50         at org.junit.Assert.fail(Assert.java:86)
14:47:50         at org.junit.Assert.assertTrue(Assert.java:41)
14:47:50         at org.junit.Assert.assertTrue(Assert.java:52)
14:47:50         at org.elasticsearch.indices.recovery.RecoveryRequestTrackerTests.testIdempotencyIsEnforced(RecoveryRequestTrackerTests.java:91)
```

Looks like this particular test has failed on 7.x a few times but was related to #57199. This may be a red herring?

**Comment by pgomulka** [comment]
failed again on 7.x https://gradle-enterprise.elastic.co/s/2l2uruhuf7tom

**Comment by cbuescher** [comment]
Another one on 7.x: https://gradle-enterprise.elastic.co/s/ffu4hqpmethdw

**Comment by tbrooks8** [comment]
Fixed by #59187.
